src/Tracy/Logger/StreamLogger.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/**
+ * This file is part of the Tracy (https://tracy.nette.org)
+ * Copyright (c) 2004 David Grudl (https://davidgrudl.com)
+ */
+
+declare(strict_types=1);
+
+namespace Tracy;
+
+
+class StreamLogger implements ILogger
+{
+
+	/** @var string path to a file where errors should be logged */
+	public string $path;
+
+
+	public function __construct(string $path)
+	{
+		$this->path = $path;
+	}
+
+
+	public function log(mixed $message, string $level = self::INFO)
+	{
+		$line = $this->formatLogLine($message);
+
+		if (!@file_put_contents($this->path, $line . PHP_EOL)) { // @ is escalated to exception
```

Comment:
you are missing `FILE_APPEND | LOCK_EX` flags

Reply:
FILE_APPEND and LOCK_EX don't work with `php://stdout`. Tested. Any ideas?

Reply:
Works for me. What is your platform and SAPI?

Reply:
```
➜ php -v
PHP 8.2.3 (cli) (built: Feb 15 2023 00:58:03) (NTS)
Copyright (c) The PHP Group
Zend Engine v4.2.3, Copyright (c) Zend Technologies
    with Zend OPcache v8.2.3, Copyright (c), by Zend Technologies
```

OSX / M1 ultra 
SAPI: php

Reply:
I would omit flags for php:// streams. These streams are already opened by php
https://www.php.net/manual/en/wrappers.php.php#wrappers.php.std

Reply:
@f3l1x Try checking `stream_isatty(STDOUT)` and run PHP with and without redirecting stdout to file (`php test > file.txt`). It should be possible to lock stdout when it is redirected to file.